consult.el

```diff
@@ -1319,6 +1318,7 @@ Prepend PREFIX in front of all items."
 
 (defun consult--flymake-candidates ()
   "Return Flymake errors as alist."
+  (require 'flymake)
```

Comment:
No. This is not good enough, since this breaks package-lint. It has to go to an extra file.

Reply:
Take two, how about using declare-function declarations, to satisfy package-lint. This is a commonly used pattern, e.g. https://github.com/emacs-mirror/emacs/blob/c6c4e746036c5ba714bfce565de13d713f3cb8b5/lisp/emacs-lisp/package.el#L1304

Reply:
Wouldn't it be easier to add an extra file like `consult-flycheck` does?

Reply:
Ignore this. I agree moving this to a separate file is a better option.

Reply:
No, this is not good enough since this breaks if flymake changes. if you are doing that you are basically circumventing static analysis.

Reply:
Done.